vql/linux/dnssnoop/dnssnoop.bpf.c

```diff
@@ -0,0 +1,66 @@
+// +build ignore
+
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2022 SUSE Linux Products GmbH. All Rights Reserved.
+//
+// Based on tcpaccept(8) from BCC by Brendan Gregg
+#include "vmlinux.h"
+
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_core_read.h>
+#include <bpf/bpf_tracing.h>
+#include <bpf/bpf_endian.h>
+
+#define PACKET_HOST 0
+#define ETH_P_IP    0x0800
+#define ETH_P_IPV6  0x86DD
+
+SEC("socket")
+int socket_filter(struct __sk_buff *skb)
+{
+	u16 sport;
+	u64 offset = 0;
+	u16 protocol;
+
+	if (skb->pkt_type != PACKET_HOST)
+		return 0;
+
+	if (bpf_skb_load_bytes(skb, offsetof(struct ethhdr, h_proto), &protocol, 2) < 0)
```

Comment:
Does this mean it ends up being ethernet-specific? That'll cover most cases, but could end up leaving e.g. VPNs uncovered.

Reply:
I think this indeed could miss i.e l3 interfaces (like vpns). But i think the problem stems from the fact the way raw sockets are implemented. Looking at the documentation for AF_PACKET: 

```
When protocol is set to htons(ETH_P_ALL), then all
       protocols are received.  All incoming packets of that protocol
       type will be passed to the packet socket before they are passed
       to the protocols implemented in the kernel.
```